C.js

```diff
@@ -1,77 +1,266 @@
+    specifyTaskList(task, taskElement);    
+};
+
+const initializeTasksFromLocalStorage = () => {
+
+    tasks.forEach(task => {
+        createTaskElementFromLocalStorage(task);
+    });
+};
+
+initializeTasksFromLocalStorage();
 
 const clearTaskInput = (event) => event.target.value = "";
 
 const newTask = (event) => {
-    const task = createNewTaskElement();
-    const checkbox = createNewCheckboxElement();
-    const taskDescription = createTaskDescription(event);
-    const deleteTaskButton = createDeleteTaskButtonElement();
-    
-    task.appendChild(checkbox);
-    task.appendChild(taskDescription);
-    task.appendChild(deleteTaskButton);
+    const task = connectTaskContent(generateTaskId(taskNumber), event.target.value);
     
     todoList.appendChild(task);
+    addTaskToLocalStorage(task);
+    updateTaskNumber();
     clearTaskInput(event);
 };
 
-const switchTaskList = (event) => {
-    const task = event.target.parentElement;
-    event.target.checked ? moveToDone(task) : moveToTodo(task);
-};
-
 const moveToDone = (task) => {
     todoList.removeChild(task);
     doneList.appendChild(task);
-    task.classList.remove("toDoTask");
-    task.classList.add("doneTask");
 };
 
 const moveToTodo = (task) => {
     doneList.removeChild(task);
     todoList.appendChild(task);
-    task.classList.remove("doneTask");
-    task.classList.add("toDoTask");
 };
 
-const deleteTask = (event) => event.target.parentElement.remove();
+const addTaskToLocalStorage = (task) => {
+    const savedTask = {
+        id: task.id,
+        description: task.children[0].children[1].value,
+        list: task.children[0].children[0].checked ? "done" : "todo"
+    }
+
+    tasks.push(savedTask);
+
+    updateTasksInLocalStorage(tasks);
+};
+
+const removeTaskFromLocalStorage = (event) => {
+    const taskId = event.target.parentElement.parentElement.id;
+
+    tasks = tasks.filter(task => task.id !== taskId);
+
+    updateTasksInLocalStorage(tasks);
+};
+
+const updateTaskDescriptionInLocalStorage = (event) => {
+    const taskId = event.target.parentElement.parentElement.id;
+
+    tasks.map(task => {
+        if (task.id === taskId) {
+            return task.description = event.target.value;
+        }
+
+        return task;
+    });
+
+    updateTasksInLocalStorage(tasks);
+};
+
+const updateTaskListInLocalStorage = (event) => {
+    const taskId = event.target.parentElement.parentElement.id;
+    
+    tasks.map(task => {
+        if (task.id === taskId) {
+            task.list = event.target.checked ? "done" : "todo";
+        }
+    });
+
+    updateTasksInLocalStorage(tasks);
+};
+
+const updateTasksInLocalStorage = (savedTasks) => localStorage.setItem("tasks", JSON.stringify(savedTasks));
+
+const RemoveAllListTasksFromLocalStorage = (listName) => {
+
+    tasks = tasks.filter(task => task.list !== listName);
+    updateTasksInLocalStorage(tasks);
+};
+
+const deleteListTasks = (list) => {

```

Comment:
give more indicative names, because you are working with 3 lists of tasks: 1 in local storage, 1 in DOM, 1 in the js script.
```suggestion
const clearTasksDisplay = (list) => {
```